content/blog/lfx-mentorship-2025-term2-zzk.md

```diff
@@ -0,0 +1,301 @@
+
+These experiences made me feel that the LFX Mentorship project about AI Gateway distributed tracing was an excellent opportunity to deepen my learning and contribute to the open source community.
+
+## Application Strategy: Contribute First, Apply Later
+
+I know everyone gets excited when they see a project they're interested in and can't wait to apply. But I adopted a different strategy: **first deeply understand the project, actively participate in the community, then submit the application**.
+
+### Deep Product Experience
+
+Instead of rushing to submit my application, I first went to actually experience the product involved in the project: [AI Gateway](https://kgateway.dev/docs/ai/about/). During usage, I discovered a practical problem - I couldn't directly access the LLM service providers mentioned in the official guide.
+
+### Proactive Problem Solving
+
+I didn't just report the problem, but proactively proposed a solution. I brought up this issue in the community Slack:
+
+{{< reuse-image src="blog/zzk-slack-first-context.png" width="750px" caption="First community interaction">}}
+
+Eventually, I successfully submitted a PR: [Allow overriding default LLM provider endpoints](https://github.com/kgateway-dev/kgateway/pull/11282), which solved this configuration issue.
+
+### Why This Approach Is Important
+
+By contributing code before applying, I achieved several goals:
+- **Deep project understanding**: Truly understood the project's architecture and pain points
+- **Building trust**: Proved to the mentor that I had the capability to complete project tasks
+- **Community integration**: Established connections with community members in advance
+
+## Project Core: Adding Distributed Tracing to AI Gateway
+
+### The Problem to Solve
+
+AI Gateway is implemented using Envoy's extproc functionality, where all requests to LLM service providers are intercepted and processed by the extension program. When system issues occur, we can only sift through massive amounts of logs for troubleshooting, which is both time-consuming and inefficient.
+
+Distributed tracing can help us:
+- **Visualize request flow**: Clearly see the complete path of requests through the system
+- **Rapid problem identification**: Quickly find issues by querying failed requests (like `attributes.http.status=400`)
+- **Performance analysis**: Understand latency at various stages
+- **Monitor LLM calls**: Track calls and performance across different LLM service providers
+
+### Project Goals
+
+Our goal was to provide complete observability support for AI Gateway:
+
+1. **Configuration flexibility**: Users can configure OpenTelemetry tracer and Span Exporter
+2. **Standardized tracing**: Follow OpenTelemetry's GenAI semantic conventions
```

Comment:
Let's link to the GenAI semantic conventions here: https://opentelemetry.io/docs/specs/semconv/gen-ai/gen-ai-spans/